Source/JavaScriptCore/dfg/DFGObjectAllocationSinkingPhase.cpp

```diff
@@ -1266,6 +1376,7 @@ class ObjectAllocationSinkingPhase : public Phase {
             ASSERT(writes.isEmpty());
             if (Node* value = heapResolve(PromotedHeapLocation(target->identifier(), exactRead))) {
                 ASSERT(!value->replacement());
+                removeArrayDependentNodesIfNeeded(node);
```

Comment:
Is it guaranteed to be safe? Isn't it possible that GetByVal / PutByVal's index is oob?

Reply:
I think so. All lowerable `GetByVal` and `PutByVal` are guarded by `arrayMode.isInBounds()` see the `GetByVal` and `PutByVal` cases in `hanldeNode`.

Reply:
Removed `removeArrayDependentNodesIfNeeded `.